sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ManifestFileCommitProtocol.scala

```diff
@@ -63,14 +65,32 @@ class ManifestFileCommitProtocol(jobId: String, path: String)
 
     if (fileLog.add(batchId, fileStatuses)) {
```

Comment:
If this line fails after writing the content, abortJob will delete the committed files. We should not remove files if this line is called.

Reply:
Nice finding! I naively think it as atomic one - write or fail, but it does more things other than just write.
We can wrap with try-catch and check whether fileLog contains these files, but given this is not a normal case and not removing files don't hurt much (either we don't guarantee), so we can clean the list and call this line.

Reply:
Just addressed.